test/InputStates-mocha.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * File: InputStatess-mocha.js
+ */
+
+var chai = require('chai');
+var should = chai.should();
+var expect = chai.expect;
+
+var InputStates = require('../components/InputStates');
+var componentFactory = require('../components/event-component-factory.js');
+var test = require('./common-test');
+
+describe("InputStates", function() {
+
+    it("should exist as a function", function() {
+        InputStates.should.exist;
+        InputStates.should.be.a('function');
+    });
+
+    it("should throw an error if no port specified", function() {
+        expect(InputStates).to.throw(Error, /Invalid arguments specified for InputState/);
+    });
+
+    it("should be undefined if no InputState has been set for a port", function() {
+       expect( InputStates('input') ).to.be.undefined;
+    });
+
+    it("should get and set state for an IIP port", function() {
+
+        // Set test input data
+        var testData = "Test State";
+        var testLm = "LM1328113669.00000000000000001";
+        var initialTestState = { data: testData, 
+                                 lm: testLm };
+        var inportName = "input";
+
+        // Set the input state for an "input" port
+        var context = InputStates( inportName, 
+                                   initialTestState );
+
+        // Context here will be the node.js context since mocha is running on that by default
```

Comment:
There is no VNI object here because I deliberately broke this check in out from the vnis in hopes of actually getting something checked in and merged.   This check in is InputState only.  No VNI exists here. 

If we keep trying to get agreement on every part of the framework at every check in, I am concerned we will fail to make any forward progress.  When I check in the VNIs, by all means, you can request that the InputState should use VNI as the context.   If you believe VNIs should be checked in without InputState first, then let me know and I'll close this PR and do VNIs separately.